lib/ansible/cli/galaxy.py

```diff
@@ -149,11 +149,6 @@ def run(self):
         
         super(GalaxyCLI, self).run()
 
-        # if not offline, get connect to galaxy api
-        if self.action in ("import","info","install","search","login","setup","delete") or \
-            (self.action == 'init' and not self.options.offline):
-            self.api = GalaxyAPI(self.galaxy)
```

Comment:
we cannot remove this as other options use it, but it might make sense to instantiate it inside those options to avoid the eternally growing list and cases here.

for example, search and all galaxy.ansible.com related actions break